### DISCUSSION TITLE
Is it possible to release my PyTermTk app as a web app

### DISCUSSION THREAD

**Discussion by SZRabinowitz** [ceccopierangiolieugenio/pyTermTk#298] [General]
Looking for something similar to the online tryit editor. 

I want to release an app as a self contained web app.

**Comment by ceccopierangiolieugenio** [comment]
Yes it is possible but I haven't formalised the procedure yet.
I can write you down the manual procedure,
recently I also tried electron and the web app works out of the box there.

**Comment by ceccopierangiolieugenio** [comment]
basically have a look at the web exporter, it is hardcoded to export "The Dumb Paint Tool" from the folder "tools/dumbPaintTool"

In theory, if you change this with the location of your app:

https://github.com/ceccopierangiolieugenio/pyTermTk/blob/a4fd0d3e7fcafdd0f2209d930ff5be1d67001810/tools/webExporterInit.sh#L104-L108

And modify the **web.ttk.package.json** with the details of your script:

https://github.com/ceccopierangiolieugenio/pyTermTk/blob/a4fd0d3e7fcafdd0f2209d930ff5be1d67001810/tools/dumbPaintTool/web.ttk.package.json#L1-L12

It should works.

**Comment by ceccopierangiolieugenio** [comment]
As example you can clone pyTermTk
and run 
```bash
tools/webExporterInit.sh
```
and check the content of the file **itchExport.zip**

![image](https://github.com/user-attachments/assets/97eb7267-580b-4a55-9231-76f09b7e50f1)

the important files are the **web.ttk.package.json** and the **bin/DPT.tgz**

You can also manually edit it and put your app as tarball inside the bin folder and adapt the json file to point to your app, you can also add as many tarballs as you want of other libraries (i.e. pygments)
